### ISSUE TITLE
xarray coordinate names are inconsistent for raw and processed data

### ISSUE THREAD

**Issue by reiserm** [European-XFEL/EXtra-data#88]
The x/y pixel coordinates of an AGIPD module are labeled `dim_0` and `dim_1` for the processed data. In case of the raw data we have the additional gain coordinate which is labeled `dim_0`. So now the pixel coordinates are `dim_1` and  `dim_2`. This is a bit inconvenient when using raw and processed data.

**Comment by takluyver** [comment]
The generic `get_array()` method doesn't try to guess anything about the dimensions, apart from the first dimension (trains). So it just gives them numbers by default. You can work around this either by using the `AGIPD1M` accessor class, which tries to make an educated guess what each dimension is:

```python
from extra_data.components import AGIPD1M
agipd = AGIPD1M(run, modules=[0])  # Use modules= to select a single module
agipd.get_array('image.data')
```

Or you can pass labels for the dimensions to the generic `get_array()` to use instead of the generic `dim_0` names:

```python
run.get_array(source, 'image.data', extra_dims=('raw_gain', 'pixel0', 'pixel1'))
```

**Comment by reiserm** [comment]
Yes, I know. I use the `get_dask_array` method instead of `get_array`, but I guess that makes no difference. Actually the trains and pulses are labeled by the coordinate names `trainId` and `pulseId`, respectively. Although both dimensions are stacked to the MultiIndex `train_pulse` first. My point is that when somebody relies on the xarray coordinates and finds that the coordinates are `trainId`, `pulseId`, `dim_0`, `dim_1`. It seems like all coordinates are named, but the names differ between raw and processed data. But anyway it is not a big deal if you know about it of course.

**Comment by tmichela** [comment]
Just checked and using the `extra_data.components`, you get labeled axis for `get_array` but not for `get_dask_array`. We should update it to label properly dask array as well.

@takluyver I was thinking for the `DataCollection` object, would it be too weird to have a mapping of known sources and use meaningful label when available or generic ones when not?

**Comment by takluyver** [comment]
I think there is/was a technical limitation in Xarray that you couldn't name only some dimensions - if you wanted to name any, you had to name them all. Maybe that's gone now and we can name the train dimension without naming the others.

I agree that `agipd.get_dask_array()` should match the dimension names from `agipd.get_array()`. I'm not so sure about identifying source names - I can definitely see the attraction, but there's also a risk that we end up giving people information that's either wrong or confusing, and the source of it is hidden in a library somewhere.

I'm also not sure how we should handle any changes to dimension names. They're effectively part of the public API - any code that uses the existing names will break if we just change them by default. I wouldn't say never, but I am keen to avoid breaking downstream code if there's any practical way to avoid it.
